## Final Project Submission

Please fill out:
* Student name: Eleanor Kelman
* Student pace: self paced
* Scheduled project review date/time: 
* Instructor name: Morgan Jones
* Blog post URL:


In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

### <font color = '#044cb3'>The datasets I will be analyzing comes from a CSV file from the National Transportation Safety Board. It comprises civil aviation accident records from 1962-2023.

In [119]:
# import dataset

df = pd.read_csv('data/Aviation_Data.csv')

df.head(3)

/Users/eleanorkelman/anaconda3/envs/learn-env/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  

[3 rows x 31 columns]

In [3]:
# see list of columns

df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

### <font color = '#044cb3'>I've listed below all the columns from the dataframe in plain English, so that we may further hone in on the information of most interest to us.

- event ID
- investigation type
- accident number
- event date
- location
- country
- latitude
- longitude
- airport code
- airport name
- injury severity
- aircraft damage
- aircraft category
- registration number
- make
- model
- amateur-built
- number of engines
- engine type
- FAR description
- schedule
- purpose of flight
- air carrier
- total fatal injuries
- total serious injuries
- total minor injuries
- total uninjured
- weather condition
- broad phase of flight
- report status
- publication date

### <font color = '#044cb3'>Let's randomly select some rows to look at them more closely and see if there is any information that immediately jumps out to us as pertinent.

In [4]:
# for real purposes, you would want to randomize these yourself; I used rng to grab 5 rows
# to examine in-depth

sample_row_1 = df.iloc[40359]
sample_row_2 = df.iloc[89396]
sample_row_3 = df.iloc[24170]
sample_row_4 = df.iloc[81282]
sample_row_5 = df.iloc[10918]

In [5]:
sample_row_1

Event.Id                  20001208X06501
Investigation.Type              Accident
Accident.Number               FTW96LA348
Event.Date                    1996-08-16
Location                    VAN HORN, TX
Country                    United States
Latitude                             NaN
Longitude                            NaN
Airport.Code                         NaN
Airport.Name                         NaN
Injury.Severity                Non-Fatal
Aircraft.damage              Substantial
Aircraft.Category                    NaN
Registration.Number                XBINF
Make                       Vans Aircraft
Model                               RV-4
Amateur.Built                        Yes
Number.of.Engines                      1
Engine.Type                Reciprocating
FAR.Description                      NaN
Schedule                             NaN
Purpose.of.flight               Personal
Air.carrier                          NaN
Total.Fatal.Injuries                   0
Total.Serious.In

### <font color = '#044cb3'>We note that *sample_row_1* is of a tiny (1 engine, 1 passenger) non-fatal crash (1 minor injury) with substantial aircraft damage out of an unrecorded, or no-name, airport. Furthermore, the weather condition is listed as VMC (visual meteorological conditions), meaning the flight was likely not visually strenuous. Furthermore, this is an amateur-built aircraft model and the crash report lists probable cause.

In [6]:
sample_row_2

Event.Id                                       20220622105301
Investigation.Type                                   Accident
Accident.Number                                    ERA22LA277
Event.Date                                         2022-06-20
Location                                       Wilmington, NC
Country                                         United States
Latitude                                              334611N
Longitude                                            0772920W
Airport.Code                                              NaN
Airport.Name                                              NaN
Injury.Severity                                       Serious
Aircraft.damage                                   Substantial
Aircraft.Category                                    Airplane
Registration.Number                                    N337AX
Make                                      Hawker Aircraft Ltd
Model                                           Hawker Hunter
Amateur.

### <font color = '#044cb3'>We note that *sample_row_2* is of a tiny (1 engine, 1 passenger) serious crash (1 minor injury) with substantial aircraft damage out of an unrecorded, or no-name, airport. Furthermore, the weather condition is listed as VMC (visual meteorological conditions), meaning the flight was likely not visually strenuous. Furthermore, this is not an amateur-built aircraft model and there is no report status.

In [7]:
sample_row_3

Event.Id                    20001213X29391
Investigation.Type                Accident
Accident.Number                LAX89DUM04B
Event.Date                      1989-09-30
Location                      EL CAJON, CA
Country                      United States
Latitude                               NaN
Longitude                              NaN
Airport.Code                           SEE
Airport.Name               GILLESPIE FIELD
Injury.Severity                  Non-Fatal
Aircraft.damage                Substantial
Aircraft.Category                      NaN
Registration.Number                 N711MH
Make                      Stolp Starduster
Model                                SA300
Amateur.Built                          Yes
Number.of.Engines                        1
Engine.Type                  Reciprocating
FAR.Description                        NaN
Schedule                               NaN
Purpose.of.flight                 Personal
Air.carrier                            NaN
Total.Fatal

### <font color = '#044cb3'>We note that *sample_row_3* is of a small (1 engine, 3 passenger) non-fatal crash (3 uninjured) with substantial aircraft damage out of Gillespie Field airport (a small airport in California [https://www.sandiegocounty.gov/content/sdc/dpw/airports/gillespie.html]). Furthermore, the weather condition is listed as VMC (visual meteorological conditions), meaning the flight was likely not visually strenuous. Furthermore, this is an amateur-built aircraft model and the crash report lists probable cause.

In [8]:
sample_row_4

Event.Id                                                     20170924X10820
Investigation.Type                                                 Accident
Accident.Number                                                  ERA17LA334
Event.Date                                                       2017-09-24
Location                                                     Darlington, SC
Country                                                       United States
Latitude                                                            034253N
Longitude                                                          0795553W
Airport.Code                                                            PVT
Airport.Name                                                           NONE
Injury.Severity                                                   Non-Fatal
Aircraft.damage                                                 Substantial
Aircraft.Category                                                  Airplane
Registration

### <font color = '#044cb3'>We note that *sample_row_4* is of a tiny (1 engine, 1 passenger) non-fatal crash (1 minor injury) with substantial aircraft damage out of a privately-owned airport. Furthermore, the weather condition is listed as VMC (visual meteorological conditions), meaning the flight was likely not visually strenuous. Furthermore, this is an amateur-built aircraft model and the crash was due to fuel starvation from a clogged fuel filter.

In [9]:
sample_row_5

Event.Id                   20001214X35661
Investigation.Type               Accident
Accident.Number                CHI85LA116
Event.Date                     1985-02-20
Location                      PONTIAC, MI
Country                     United States
Latitude                              NaN
Longitude                             NaN
Airport.Code                          PTK
Airport.Name              OAKLAND-PONTIAC
Injury.Severity                 Non-Fatal
Aircraft.damage               Substantial
Aircraft.Category                     NaN
Registration.Number                N6450K
Make                               Cessna
Model                                 172
Amateur.Built                          No
Number.of.Engines                       1
Engine.Type                 Reciprocating
FAR.Description                       NaN
Schedule                              NaN
Purpose.of.flight           Instructional
Air.carrier                           NaN
Total.Fatal.Injuries              

### <font color = '#044cb3'>We note that *sample_row_5* is of a tiny (1 engine, 1 passenger) non-fatal crash (1 uninjured) with substantial aircraft damage out of Oakland County International Airport (a mid-sized airport in Michigan [https://www.oakgov.com/community/airports/oakland-county-international-airport]). Furthermore, the weather condition is listed as VMC (visual meteorological conditions), meaning the flight was likely not visually strenuous. Furthermore, this is not an amateur-built aircraft model and the crash report lists probable cause.

### <font color = '#044cb3'> Entirely based on the samples (which are only a few out of many, many rows), we can narrow down our conclusions to certain rows. To start, let's see if there exists a large discrepancy between the number of amateur and professional crashed aircrafts.

In [35]:
# aggregate data by Amateur.Built column

total_builds = df['Amateur.Built'].value_counts()

# check if there are any non-Boolean columns to ensure comprehensive agreggation

sum(total_builds) == df['Amateur.Built'].count()

True

In [38]:
# break up data into amateur and professional builds

builds_amateur = total_builds['Yes']

builds_prof = total_builds['No']

In [39]:
# calculate percentage of amateur and professional builds

percent_amateur = (builds_amateur / sum(total_builds)) * 100

percent_prof = (builds_prof / sum(total_builds)) * 100

print(f"The percentage of amateur builds is {percent_amateur}%.")

print(f"The percentage of professional builds is {percent_prof}%.")

The percentage of amateur builds is 9.545316318830459%.
The percentage of professional builds is 90.45468368116954%.


### <font color = '#044cb3'>There are far more professional builds listed, but we cannot necessarily glean information from this. However, we could hypothesize that a professional plane, by nature of flying higher, at faster speeds, and with more engines, could lead to a more severe crash. Let's break down these builds into the severity of the crashes.

In [42]:
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [77]:
df['Injury.Severity'].value_counts()

Non-Fatal     67357
Fatal(1)       6167
Fatal          5262
Fatal(2)       3711
Incident       2219
              ...  
Fatal(80)         1
Fatal(31)         1
Fatal(132)        1
Fatal(228)        1
Fatal(144)        1
Name: Injury.Severity, Length: 109, dtype: int64

### <font color = '#044cb3'>Looking at the injury severities, there appear to be a lot of names to mean the same thing (for example, the values of *Fatal*, *Fatal(1)*, and *Fatal(228)* all refer to deadly crashes). Let's start by cleaning up this data so that there are only a few specific injury types.

In [120]:
# clean up list of fatal crashes to retrieve all rows

fatal_rows = ['Fatal']

unique_values = set(list(df['Injury.Severity']))

for value in unique_values:
    if type(value) == str:
        if value != 'Non-Fatal' and 'Fatal' in value:
            fatal_rows.append(value)

In [132]:
df['Injury.Severity'] = df['Injury.Severity'].apply(lambda value: "Fatal" if value in fatal_rows else value)

In [137]:
df['Injury.Severity'].value_counts()

Non-Fatal      67357
Fatal          17826
Incident        2219
Minor            218
Serious          173
Unavailable       96
Name: Injury.Severity, dtype: int64

### <font color = '#044cb3'>Now let's tally up the fatal/serious injuries in comparison to the minor/non-fatal incidents. Although there are two other types, incident and unavailable, there are very few rows of the latter and the former is not as clearly-defined as a type of severity.
    
### <font color = '#044cb3'>Let's also be sure to consider these in terms of amateur and professional crafts.

In [160]:
total_amateur_serious = df_amateur[df_amateur['Injury.Severity'] == 'Fatal']['Event.Id'].count() + df_amateur[df_amateur['Injury.Severity'] == 'Serious']['Event.Id'].count()
total_amateur_minor = df_amateur[df_amateur['Injury.Severity'] == 'Non-Fatal']['Event.Id'].count() + df_amateur[df_amateur['Injury.Severity'] == 'Minor']['Event.Id'].count()

total_prof_serious = df_prof[df_prof['Injury.Severity'] == 'Fatal']['Event.Id'].count() + df_prof[df_prof['Injury.Severity'] == 'Serious']['Event.Id'].count()
total_prof_minor = df_prof[df_prof['Injury.Severity'] == 'Non-Fatal']['Event.Id'].count() + df_prof[df_prof['Injury.Severity'] == 'Minor']['Event.Id'].count()

In [161]:
percent_amateur_serious = (total_amateur_serious / df['Injury.Severity'].count()) * 100
percent_amateur_minor = (total_amateur_minor / df['Injury.Severity'].count()) * 100

percent_prof_serious = (total_prof_serious / df['Injury.Severity'].count()) * 100
percent_prof_minor = (total_prof_minor / df['Injury.Severity'].count()) * 100

print(f"The percentage of serious amateur crashes is {percent_amateur_serious}")
print(f"The percentage number of minor amateur crashes is {percent_amateur_minor}")

print(f"The percentage number of serious professional crashes is {percent_prof_serious}")
print(f"The percentage number of minor professional crashes is {percent_prof_minor}")

The percentage of serious amateur crashes is 5.36130801351705
The percentage number of minor amateur crashes is 69.9678002935521
The percentage number of serious professional crashes is 0.8226285428210584
The percentage number of minor professional crashes is 6.901887608233112


In [ ]:
df_amateur = df[df['Amateur.Built'] == 'No']

df_prof = df[df['Amateur.Built'] == 'Yes']